### PR TITLE
Use conda for all dependencies.

### PR DESCRIPTION
**Pull request by trivialfis**
For consistency and performance.

I use `mamba` for creating environments, which is much faster than the resolver from pip.

### CODE REVIEW

File: scripts/generate-conda-envs.py [hunk 1]
```diff
@@ -485,7 +488,9 @@ def section_selected(section):
             use=config.use,
             python=config.python,
             conda_sections=conda_sections,
-            pip=PIP_TEMPLATE.format(pip_sections=pip_sections),
+            pip=PIP_TEMPLATE.format(pip_sections=pip_sections)
+            if pip_sections
+            else "",
```
Comment (bryevdv) [inline scripts/generate-conda-envs.py:493]: I think it would be better to handle this in the template, to keep all the cases here consistent

**Comment by bryevdv** [comment]
> better suited for pip sourcing

@manopapad No special concerns from me.  If they were put in `pip` section originally then that's because there was some issue obtaining then via `conda` at the time. If the issues are now resolved then switching to using `conda` seems fine. (Would just acknowledge that if anything ever becomes difficult to obtain via conda it might need to get moved to `pip`  in general).

Just one comment regarding moving some logic to the jinja template

File: scripts/generate-conda-envs.py (hunk 1 above)
Comment (trivialfis) [reply scripts/generate-conda-envs.py]: Thank you for the suggestion, reverted the change.

**Comment by trivialfis** [comment]
I have tested with `mamba env update -f`, but your suggestion is better, I will omit the pip section.

**Comment by trivialfis** [comment]
Looking into the code, I think the generator doesn't depend on jinja2, actually, it doesn't have any dependency since it's the script used to list dependencies in the first place. I think we might want to stay with an empty section or use conditions in the Python format function.

**Comment by bryevdv** [comment]
Oh right we had used jinja originally but then removed it, and I misled myself with the `_TEMPLATE` name. In that case yes your original code seems appropriate.

**Comment by manopapad** [comment]
So this is good to merge then?

**Comment by bryevdv** [comment]
No I think the going back to 
```
pip=PIP_TEMPLATE.format(pip_sections=pip_sections)
            if pip_sections
            else "",
```
is probably the best idea.

**Comment by trivialfis** [comment]
> No I think the going back to the original

Done.

**Review by bryevdv — APPROVED** [review]
LGTM @manopapad

**Pull request merged**